src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignerInfoTests.netcoreapp.cs

```diff
@@ -77,6 +77,51 @@ public static void SignerInfo_RemoveUnsignedAttribute_RemoveCounterSignature()
             Assert.Equal(secondSignerCounterSignature, cms.SignerInfos[0].CounterSignerInfos[0].GetSignature());
         }
 
+        [Fact]
+        public static void SignerInfo_RemoveUnsignedAttributeFromCounterSigner_Removes()
+        {
+            SignedCms cms = new SignedCms();
+            cms.Decode(SignedDocuments.OneRsaSignerTwoRsaCounterSigners);
+
+            AsnEncodedData attribute1 = CreateTimestampToken(1);
+            AsnEncodedData attribute2 = CreateTimestampToken(2);
+
+            SignerInfo counterSigner = cms.SignerInfos[0].CounterSignerInfos[0];
+            counterSigner.AddUnsignedAttribute(attribute1);
+            counterSigner.AddUnsignedAttribute(attribute2);
+
+            ReReadSignedCms(ref cms);
+
+            // attributes got sorted, therefore number changed
+            Assert.Equal(1, cms.SignerInfos[0].CounterSignerInfos[1].UnsignedAttributes.Count);
+            Assert.Equal(2, cms.SignerInfos[0].CounterSignerInfos[1].UnsignedAttributes[0].Values.Count);
+
+            counterSigner = cms.SignerInfos[0].CounterSignerInfos[1];
+            counterSigner.RemoveUnsignedAttribute(counterSigner.UnsignedAttributes[0].Values[0]);
+
+            // We didn't modify current instance
+            Assert.Equal(1, counterSigner.UnsignedAttributes.Count);
+            Assert.Equal(2, counterSigner.UnsignedAttributes[0].Values.Count);
```

Comment:
After the first ReRead (when the document is now in the "ready to test" state), save cms.SignerInfos[0], and see that that instance behaves as expected (either does, or doesn't, see the changes).